### ISSUE TITLE
Micro-Averaging for AUC and PR curves

### ISSUE THREAD

**Issue by maxwell-geospatial** [tidymodels/yardstick#448]
I had a question relating to the yardstick package and multi-class assessment more generally. 

For a multi-class problem, macro-averaged, class-aggregated recall, precision, and F1-score are equivalent and actually identical to overall accuracy. In contrast, macro-averaged, class-aggregated recall, precision, and F1-score are unique metrics that provide new information. 

We are trying to understand how this relates to calculating ROC and Precision-Recall curves and the associated area under the curve metric. In the yardstick package, you have implemented “binary”, “hand_till”, “macro”, and “macro_weighted” averaging for ROC curves and “binary”, “macro”, and “macro_weighted” for precision-recall curves. There is no option for “micro” averaging. 

Given that micro-averaged, class-aggregated precision, recall, and F1-score are equivalent, it seems that this issue would extend to ROC and P-R curves since ROC curves make use of recall and recall relative to the “background” class and P-R curves make use of precision and recall. 

In short, it seems that it does not make sense to use micro-averaging when producing ROC and precision-recall curves and the associated AUC metric. We are assuming this is why you did not include this option in the yardstick package. However, references to micro-averaging are made in other sources. For example, the multiROC package states that “Micro-average ROC/AUC was calculated by stacking all groups together, thus converting the multi-class classification into binary classification. Macro-average ROC/AUC was calculated by averaging all groups results (one vs rest) and linear interpolation was used between points of ROC.” Although this makes sense at first glance, it seems that micro-averaging is fundamentally flawed and is not a legitimate means to produce multi-class curve sets. Specifically, at any given probability threshold, micro-averaged, class-aggregated recall and precision are equivalent, so there would be no value in plotting these to separate axes. 

Would you be willing to offer any insights here? There seems to be some confusion in the existing literature………

**Comment by EmilHvitfeldt** [comment]
Hello @maxwell-geospatial 👋 

Have you seen the vignette on [Multiclass averaging](https://yardstick.tidymodels.org/articles/multiclass.html)? it might shed some light on what we mean with the different terms n this package.

**Comment by maxwell-geospatial** [comment]
I have looked at that vignette. I am not sure that it answers this specific question. I’m trying to determine whether micro-averaging actually makes sense for ROC and PR curves. In the yardstick package, micro-averaging is not an option. I feel that micro-averaging doesn’t make sense for the reasons stated in the post. I was curious if in developing yardstick, you came across a similar issue and intentionally decided not to include a micro-averaging option.

**Comment by EmilHvitfeldt** [comment]
we haven't implemented `"micro"` for these AUC metrics for a couple of reasons. Firstly it is quite hard to make the math work out. Secondly, in the beginning we used scikit-learn as a reference to decide what metrics and how they should be calculated. 

We saw that the scikit-learn metrics maintainer was like "micro makes no sense" and we have followed suit and not implemented it https://github.com/scikit-learn/scikit-learn/issues/3298#issuecomment-251713956
